### PR TITLE
feat: typescript conversion AL-2148

### PR DESCRIPTION
**Pull request by frederic-schwarz**
converted `csg.js` to typescript
* using both namespace `CSG` and class `CSG` to maintain the current interface
* removes the need to maintain separate csg.d.ts

`./dist/csg.js` and `./dist/csg.d.ts`
 are generated on `npm run build`

### CODE REVIEW

File: csg.ts [hunk 1]
```diff
@@ -185,8 +186,11 @@ CSG.cube = function(options) {
   ].map(function(info) {
     return new CSG.Polygon(info[0].map(function(i) {
       var pos = new CSG.Vector(
+        //@ts-ignore
         c.x + r[0] * (2 * !!(i & 1) - 1),
+        //@ts-ignore
         c.y + r[1] * (2 * !!(i & 2) - 1),
+        //@ts-ignore
         c.z + r[2] * (2 * !!(i & 4) - 1)
```
Comment (frederic-schwarz) [inline csg.ts]: that part `!!(i & 1)` from the initial code works in javascript: `2 * false - 1 = -1` but typescript complains
add `//@ts-ignore` to make it transpile

Comment (stavros-papadopoulos) [reply csg.ts]: How about this?

```
let dX = i & 1 ? r[0] : -r[0]
let dY = i & 2 ? r[1] : -r[1]
let dZ = i & 4 ? r[2] : -r[2]
let pos = new CSG.Vector(c.x + dX, c.y + dY, c.z + dZ)
```

This is why I dislike 'clever' code by the way :). Sooner or later, something like TypeScript will complain.

**Comment by stavros-papadopoulos** [comment]
Yeah, I think converting to TypeScript is the way to go!

**Review by stavros-papadopoulos — APPROVED** [review]

File: csg.ts (hunk 1 above)
Comment (frederic-schwarz) [reply csg.ts]: great! fc10b2302107d5039e91ddf6837094957ac4de87

**Pull request merged**